### PR TITLE
feat: open prop will override any event handler

### PR DESCRIPTION
**Pull request by nelsonchen90**
To address this: https://github.com/carbon-design-system/carbon-components-react/issues/669#issuecomment-370884282 
Closes carbon-design-system/carbon-components-react#669

Tooltip will be fully controlled by `open` if it's defined.

#### Changelog

**Changed**

* if `props.open` is defined on the Tooltip, then all event handler will be skipped /overridden. Changing the `open` from prop is the only way to toggle the display of the tooltip

* Tooltip `state.open` will be initialized to `false` if `open` on props is undefined.

**Removed**

* removed the default prop value from the Tooltip component.

### CODE REVIEW

File: src/components/Tooltip/Tooltip.js [hunk 1]
```diff
@@ -142,7 +141,7 @@ export default class Tooltip extends Component {
   };
 
   state = {
-    open: this.props.open,
+    open: this.props.open === undefined ? false : this.props.open,
```
Comment (asudoh) [inline src/components/Tooltip/Tooltip.js]: Would we anticipate non-boolean and non-null/undefined values for `state.open`? If not, wondering if we can do `Boolean(this.props.open)`.

**Comment by nelsonchen90** [comment]
@asudoh Thanks for reviewing.

I am trying to implement a 'popover' type of tooltip which user can hover on the tootlip content. In order to do that, I need a wrapper around the tooltip and that wrapper is going to control the tooltip thru the `open` prop.

if we don't ignore the other events when the `open` is defined (true/false), those events such as `mouseout` will not honor the value of `open` and close the tooltip instead of keeping it open.

Let me know if this use case make sense. Thanks again!

File: src/components/Tooltip/Tooltip.js (hunk 1 above)
Comment (nelsonchen90) [reply src/components/Tooltip/Tooltip.js]: I will make the code change. thanks for pointing our

**Comment by nelsonchen90** [comment]
@asudoh Yes, I think that's exactly what we wanted. Maybe we can just close this PR then. I think your implementation makes things a lot easier!

**Comment by asudoh** [comment]
Thanks @nelsonchen90! Will port https://github.com/carbon-design-system/carbon-components/pull/639 to this repo.

**Comment by tw15egan** [comment]
Closing PR in favor of https://github.com/carbon-design-system/carbon-components-react/pull/696

**Pull request closed without merging**